### PR TITLE
APIS-4747 - Add access requirements functionality

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: app/uk/gov/hmrc/apisubscriptionfields/model/test.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"access": {
+    "devhubAccessLevels": {
+      "readWrite" : "NoOne",
+      "readOnly": "Admin"
+    }
+    "gateKeeperAccessLevels": {
+      "readWrite" : "SuperUser",
+      "readOnly": "User"
+    }
+}
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apisubscriptionfields/model/test.json:10]: Newline pls

File: test/uk/gov/hmrc/apisubscriptionfields/FieldDefinitionTestData.scala [hunk 2]
```diff
@@ -44,10 +44,11 @@ trait FieldsDefinitionTestData extends TestData {
...
   final val FakeFieldDefinitionString = FieldDefinition(fieldN(2), "desc2", "hint2", FieldDefinitionType.STRING, "short description", Some(FakeValidation))
+  final val FakeFieldDefinitionWithAccess: FieldDefinition = FakeFieldDefinitionString.copy(validation = None, access = AccessRequirements(devhub = DevhubAccessRequirements(readOnly = Admininstator)))
   final val FakeFieldDefinitionSecureToken = FieldDefinition(fieldN(3), "desc3", "hint3", FieldDefinitionType.SECURE_TOKEN, "short description", Some(FakeValidation))
-  final val FakeFieldsDefinitions = NonEmptyList.fromListUnsafe(List(FakeFieldDefinitionUrl, FakeFieldDefinitionString, FakeFieldDefinitionSecureToken))
-  final val FakeFieldsDefinition = FieldsDefinition(fakeRawContext, fakeRawVersion, FakeFieldsDefinitions)
-  final val FakeFieldsDefinitionResponse = FieldsDefinitionResponse(fakeRawContext, fakeRawVersion, FakeFieldsDefinition.fieldDefinitions)
+  final val NelOfFieldDefinitions = NonEmptyList.fromListUnsafe(List(FakeFieldDefinitionUrl, FakeFieldDefinitionString, FakeFieldDefinitionSecureToken))
```
Comment (peteslater-ee) [inline test/uk/gov/hmrc/apisubscriptionfields/FieldDefinitionTestData.scala:49]: .fromListUnsafe can throw an IllegalArgumentException which is a bit nasty.
Any chance we should use the `safe` version, .fromList, and deal with the returned Option?
I know it's _only_ a test but it's still grim and sends the wrong message.

File: acceptance/uk/gov/hmrc/apisubscriptionfields/acceptance/SubscriptionFieldDefinitionsHappySpec.scala [hunk 3]
```diff
@@ -117,14 +117,14 @@ class SubscriptionFieldDefinitionsHappySpec extends AcceptanceTestSpec
       Then(s"a response with a 200 status is received")
       result shouldBe 'defined
       val resultFuture = result.value
-
+ApiFieldDefinitionsResponse
```
Comment (peteslater-ee) [inline acceptance/uk/gov/hmrc/apisubscriptionfields/acceptance/SubscriptionFieldDefinitionsHappySpec.scala:120]: What is this?

File: test/uk/gov/hmrc/apisubscriptionfields/FieldDefinitionTestData.scala (hunk 2 above)
Comment (AndySpaven) [reply test/uk/gov/hmrc/apisubscriptionfields/FieldDefinitionTestData.scala:49]: No.  The fromListUnsafe is designed to be used in just this sort of situation where you have a list that is clearly not empty.

File: app/uk/gov/hmrc/apisubscriptionfields/model/test.json (hunk 1 above)
Comment (AndySpaven) [reply app/uk/gov/hmrc/apisubscriptionfields/model/test.json:10]: Done

**Pull request merged**